lib/core/src/sdk.rs

```diff
@@ -735,48 +762,94 @@ impl LiquidSdk {
     ) -> Result<SendPaymentResponse, PaymentError> {
         self.ensure_is_started().await?;
 
-        self.ensure_send_is_not_self_transfer(&req.invoice)?;
-        self.validate_invoice(&req.invoice)?;
+        match &req.parsed_destination {
+            InputType::LiquidAddress {
+                address: liquid_address_data,
+            } => {
+                // We default to paying the invoice encoded within the BIP21
+                // if it is present
+                if let Some(invoice) = &liquid_address_data.invoice {
+                    return self.check_send_payment_invoice(invoice, req.fees_sat).await;
+                }
+
+                let Some(amount_sat) = liquid_address_data.amount_sat else {
```

Comment:
As bip21 won't necessaryily have an amount perhaps we need to change the send_payment to receive a different argument?
`send_payment(SendPaymentRequest)`

Where

```
SendPaymentRequest {
 prepare_response: PrepareSendResponse
 amount_sat: Option<u64>
}
```

Reply:
Oh, I see we thought of the same alternative. Then I agree, will fix this way